pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -91,12 +99,17 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	err = r.reconcile(ctx, c)
 	if err != nil {
 		logger.Info("Error reconciling Channel", zap.Error(err))
+		r.recorder.Eventf(c, corev1.EventTypeWarning, channelReconcileFailed, "Failed to reconcile Channel: %v", err)
```

Comment:
Is it possible to get here without recording an event inside `r.reconcile`? I'm wondering if this event will simply be redundant with an event emitted inside the function (and likely with a more specific message).